### ISSUE TITLE
Build Errors

### ISSUE THREAD

**Issue by muhammadfraz** [derzu/BodySkeletonTracker#9]
Hi,
Thank you for creating and sharing your work.
I've been trying to build it for a quick test but I've got a few error that I am not sure how to resolve considering all the paths seem correct.

/usr/bin/ld: cannot find -lnppial
/usr/bin/ld: cannot find -lnppicc
/usr/bin/ld: cannot find -lnppicom
/usr/bin/ld: cannot find -lnppidei
/usr/bin/ld: cannot find -lnppif
/usr/bin/ld: cannot find -lnppig
/usr/bin/ld: cannot find -lnppim
/usr/bin/ld: cannot find -lnppist
/usr/bin/ld: cannot find -lnppisu
/usr/bin/ld: cannot find -lnppitc
/usr/bin/ld: cannot find -llib64

collect2: error: ld returned 1 exit status
CommonCppMakefile:139: recipe for target 'Bin/x64-Release/BodySkeletonTracker' failed
make: *** [Bin/x64-Release/BodySkeletonTracker] Error 1

I am working on Ubuntu 16.04 and using Primesense sensor. The sensor works fine otherwise which means OpenNI2 is installed correctly.
Can you please help?

Thanks, 
Fraz

**Comment by derzu** [comment]
Hi,

I'm not sure about what is happening. But I think is something related to the Cuda Library. I do not use Cuda in this project.

Maybe you compiled your OpenCV activating Cuda module, did you? You can try 2 solutions:

1 - Recompile OpenCV with Cuda module.

2 - Add the Cuda library path to the compiler, you can do that changing the Makefile, adding:
CFLAGS += -L/usr/local/cuda-6.5/lib

Set you cuda path appropriately.

Tell me if it works.

Best regards,

Derzu.

**Comment by muhammadfraz** [comment]
Hi Derzu, 

Thanks for prompt reply. I already managed to find my way through the problem. You are right I compiled OpenCV with cuda support and these linking errors were caused because the makefile script were unable to find the NVIDIA performance primitives. 

So, the work around that I did was adding these paths to the LIB_DIRS_OPTIONS in CommonCppCMakefile, basically providing the path where these libraries are located:

The modified section looks like this:

LIB_DIRS_OPTION = $(foreach dir,$(LIB_DIRS),-L$(dir)) 
LIB_DIRS_OPTION += -L/usr/local/cuda -L/usr/local/cuda/lib64
LIB_DIRS_OPTION += -L$(OUT_DIR)

I believe this is not the elegant way but I was unable to find where the LIB_DIRS variable is being set, the cmake files are not easy to follow here, especially for someone like me who is not very good with cmake.

Also, the last ld error which another person faced and posted in the other issue:
/usr/bin/ld: cannot find -llib64

To me this seems like a bug as there is no lib64 static or dynamic library at least in Ubuntu16.04, so what I did was, modified the opencv.pc file located at the path: /usr/local/lib/pkgconfig/opencv.pc

I removed "-llib64" from the script and then everything build and ran successfully.

Thanks, 
Fraz
